### DISCUSSION TITLE
SWI BIDS

### DISCUSSION THREAD

**Discussion by benjamin-zimmerman** [QSMxT/QSMxT#17] [Q&A] [answered]
I'm having some trouble getting my multi-echo SWI to run through this pipeline. The BIDS folder looks strange from the bidsifying step in this code, and I was thinking of doing it manually to try to test the QSM. Can you be more specific about DICOM structure that is expected for the bidsifying step, and also an example of what the BIDS folder should look like for QSM to work? As far as I know, the BIDS extension for SWI isn't complete yet, right? So it's hard for me to know the expectation for naming/structure.

**Comment by stebo85** [comment]
Dear @benjamin-zimmerman,

yes, the BIDS standard for QSM is not final, so we currently expect the files to be like this:
{subject}/{session}/anat/*qsm*{run}*magnitude*nii*
{subject}/{session}/anat/*qsm*{run}*phase*nii*

To keep it all flexible these file patterns can be changed via command line options of run_2_qsm.py:
    parser.add_argument(
        '--subject_pattern',
        default='sub*',
        help='Pattern used to match subject folders in bids_dir'
    )

    parser.add_argument(
        '--session_pattern',
        default='ses*',
        help='Pattern used to match session folders in subject folders'
    )

    parser.add_argument(
        '--magnitude_pattern',
        default='{subject}/{session}/anat/*qsm*{run}*magnitude*nii*',
        help='Pattern to match magnitude files within the BIDS directory.'
    )

    parser.add_argument(
        '--phase_pattern',
        default='{subject}/{session}/anat/*qsm*{run}*phase*nii*',
        help='Pattern to match phase files for qsm within session folders.'
    )

Could you send us one DICOM of the phase and magnitude so we can see if we can adjust our bids converter for your data?

**Reply by stebo85** [reply]
in addition: the dicom structure does not matter as we are sorting it based on this bidsmap:
https://github.com/QSMxT/QSMxT/blob/master/bidsmap.yaml

The bids conversion should work if we change these patterns to match your particular scans :)

**Comment by benjamin-zimmerman** [comment, marked as answer]
Hi, I think I can send you a DICOM of the phase and magnitude - where/with-whom should I share it? For my purposes, though, I'll eventually be starting with BIDS data - I'm just testing right now to choose how to design our pipeline. I guess a more primary question is whether QSMxT should work with multiecho data? The echos might not automatically stack in the dcm2nii step.

**Reply by stebo85** [reply]
you can send the dicom data to me (s.bollmann@uq.edu.au) and I can change the bidsmap for you, then the conversion from dicom to bids should work.

Starting with bids data should also work nicely by just changing the file matching patterns of run_2_qsm.py

Yes, QSMxT will work well for multiecho data - you can either specify how many echoes you have or it will grab all echoes automatically:
  parser.add_argument(
        '--num_echoes',
        dest='num_echoes_to_process',
        default=None,
        type=int,
        help='The number of echoes to process; by default all echoes are processed.'
    ) 

We can also have a look at your data together in a quick zoom session if you like :)

Cheers
Steffen
